lib/identity_job_log_subscriber.rb

```diff
@@ -179,6 +218,14 @@ def trace_id(job)
   def duplicate_cron_error?(ex)
     ex.is_a?(ActiveRecord::RecordNotUnique) && ex.message.include?('(cron_key, cron_at)')
   end
+
+  def should_error?(job, ex)
+    return true if ex.nil?
+
+    job.warning_error_classes.none? { |warning_class|
```

Comment:
is `job` an instance of a job, or a job class?

Reply:
Oh it's a job. Let me relook at this once tests pass. I think there's something wrong in way we do it now.